File: java-solver/src/fi/spektrum/sudoku/dlx/ColumnNode.java
package fi.spektrum.sudoku.dlx;

/**
 *
 * @author Lasse Lybeck
 */
public class ColumnNode extends Node {

    private int size;
    private int name;

    public int getSize() {
        return size;
    }

    public void setSize(int size) {
        this.size = size;
    }

    public int getName() {
        return name;
    }

    public void setName(int name) {
        this.name = name;
    }

    @Override
    public ColumnNode getRight() {
        return (ColumnNode) super.getRight();
    }

    @Override
    public ColumnNode getLeft() {
        return (ColumnNode) super.getLeft();
    }

    void decreaseSize() {
        --size;
    }
    
    void increaseSize() {
        ++size;
    }
}

File: cpp-solver/column_node.cpp
/* 
 * File:   column.cpp
 * Author: Lasse Lybeck
 * 
 * Created on October 16, 2014, 12:46 PM
 */

#include "column_node.h"

namespace dlx {

    column_node::column_node() : size_(0) { }

    int& column_node::size() {
        return size_;
    }

    column_node*& column_node::right() {
        return (column_node*&) node::right();
    }

    column_node*& column_node::left() {
        return (column_node*&) node::left();
    }

    int& S(column_node* c) {
        return c->size();
    }

    column_node*& R(column_node* c) {
        return c->right();
    }

    column_node*& L(column_node* c) {
        return c->left();
    }

    column_node*& C(node* n) {
        return (column_node*&) n->column();
    }
}
